### PR TITLE
lightning: add option to control pause scheduler scope

### LINKED ISSUE #43657: lightning might stuck at split/scatter region
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
see https://github.com/tikv/pd/issues/6426. right now lightning pause schedule by key range, but current impl of `pd` might cause a region never been able to split successfully on some case.

we need a way to fallback to old way of pausing scheduler, i.e. by remove schedulers

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #43657

Problem Summary:

### What is changed and how it works?
- add option `pause-pd-scheduler-scope` to control the scope when pause scheduler, default=table

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - manually change CanPauseSchedulerByKeyRange to return false and set pause-pd-scheduler-scope=table, it should fail
7f6e0c05-b0dc-4ac2-92c1-0ed2b818bc18

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
add option `pause-pd-scheduler-scope` to control the scope when pause scheduler
```

### CODE REVIEW

**Review by gozssky — APPROVED** [review]

File: br/tests/lightning_csv/run.sh [hunk 1]
```diff
@@ -39,13 +41,24 @@ for BACKEND in tidb local; do
   check_contains 'id: 3'
   run_sql 'SELECT id FROM csv.empty_strings WHERE b <> ""'
   check_not_contains 'id:'
+}
 
-done
+rm -rf $TEST_DIR/lightning.log
+run_with "local" "tests/$TEST_NAME/config-pause-global.toml"
+grep -Fq 'pause pd scheduler of global scope' $TEST_DIR/lightning.log
```
Comment (lance6716) [inline br/tests/lightning_csv/run.sh:48]: What's the meaning of these checks, should it assert "global" is found and "table" is not found?

Comment (D3Hunter) [reply br/tests/lightning_csv/run.sh:48]: grep will fail if it cannot find it, not that explicit, copied from https://github.com/pingcap/tidb/blob/cde395dc597cab4540fe7aa141c5804ca8983519/br/tests/lightning_add_index/run.sh#L53

File: br/tests/lightning_csv/run.sh [hunk 2]
```diff
@@ -39,13 +41,24 @@ for BACKEND in tidb local; do
...
+! grep -Fq 'pause pd scheduler of global scope' $TEST_DIR/lightning.log
```
Comment (lance6716) [inline br/tests/lightning_csv/run.sh:49]:
```suggestion
! grep -Fq 'pause pd scheduler of table scope' $TEST_DIR/lightning.log
```

File: br/tests/lightning_csv/run.sh [hunk 3]
```diff
@@ -39,13 +41,24 @@ for BACKEND in tidb local; do
...
+
+rm -rf $TEST_DIR/lightning.log
+run_with "local" "tests/$TEST_NAME/config.toml"
+! grep -Fq 'pause pd scheduler of global scope' $TEST_DIR/lightning.log
+grep -Fq 'pause pd scheduler of global scope' $TEST_DIR/lightning.log
```
Comment (lance6716) [inline br/tests/lightning_csv/run.sh:54]:
```suggestion
grep -Fq 'pause pd scheduler of table scope' $TEST_DIR/lightning.log
```

File: br/tests/lightning_csv/run.sh (hunk 1 above)
Comment (D3Hunter) [reply br/tests/lightning_csv/run.sh:48]: shit, local run succeed

Comment (D3Hunter) [reply br/tests/lightning_csv/run.sh:48]:
```
              -e      Exit immediately if a pipeline (which may consist of a single simple command), a list, or a compound command (see SHELL GRAMMAR above), exits with a non-zero status.  The shell
                      does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test following the if or elif reserved words, part
                      of any command executed in a && or || list except the command following the final && or ||, any command in a pipeline but the last, or if the command's return value is being
                      inverted with !.
```

> **The shell does not exit if ...... the command's return value is being inverted with !**

that means some integration test's check is not valid

**Comment by D3Hunter** [comment]
ci pass
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/12563/pipeline

**Comment by D3Hunter** [comment]
@lance6716

**Review by lance6716 — APPROVED** [review]

**Pull request merged**